test/lib/whoami.js

```diff
@@ -0,0 +1,34 @@
+const { test } = require('tap')
+const requireInject = require('require-inject')
+
+test('whoami', (t) => {
+  t.plan(3)
+  const whoami = requireInject('../../lib/whoami.js', {
+    '../../lib/utils/get-identity.js': () => Promise.resolve('foo'),
+    '../../lib/npm.js': { flatOptions: {} },
+    '../../lib/utils/output.js': (output) => {
+      t.equal(output, 'foo', 'should output the username')
+    }
+  })
+
+  whoami([], (err) => {
+    t.ifError(err, 'npm whoami')
+    t.ok('should successfully print username')
+  })
+})
+
+test('whoami json', (t) => {
+  t.plan(3)
+  const whoami = requireInject('../../lib/whoami.js', {
+    '../../lib/utils/get-identity.js': () => Promise.resolve('foo'),
+    '../../lib/npm.js': { flatOptions: { json: true } },
+    '../../lib/utils/output.js': (output) => {
+      t.equal(output, '"foo"', 'should output the username as json')
+    }
+  })
```

Comment:
just a heads up since this is a very simplistic module it doesn't matter that much but in other tests we have been trying to only have a single `requireInject()` import at the top and mutate the mocked values within each test since the requireInject calls can be really slow and add precious seconds to our unit tests, e.g: https://github.com/npm/cli/pull/1786/files#diff-4b5283ecd4a834e470103ca4fc13fbd7R41

Reply:
ah! ok, i wasn't sure since i saw it both ways. tracking global state in a test felt weird so i went the direction that didn't require that, but it makes sense